src/unix/fs.c

```diff
@@ -58,6 +59,41 @@
 # include <sys/sendfile.h>
 #endif
 
+#if defined(__APPLE__)
+# include <sys/attr.h>
+
+#define SETATTRLIST                                                            \
+  struct attrlist attr_list;                                                   \
+  struct timespec times[3];                                                    \
+  unsigned i = 0;                                                              \
+                                                                               \
+  memset(&attr_list, 0, sizeof(attr_list));                                    \
+                                                                               \
+  attr_list.bitmapcount = ATTR_BIT_MAP_COUNT;                                  \
+                                                                               \
+  if (!isnan(req->btime)) {                                                    \
+    attr_list.commonattr |= ATTR_CMN_CRTIME;                                   \
+                                                                               \
+    times[i++].tv_sec = req->btime;                                            \
+    times[i].tv_nsec = (unsigned long)(req->btime * 1000000) % 1000000 * 1000; \
+  }                                                                            \
+                                                                               \
+  if (!isnan(req->atime)) {                                                    \
+    attr_list.commonattr |= ATTR_CMN_ACCTIME;                                  \
+                                                                               \
+    times[i++].tv_sec = req->atime;                                            \
+    times[i].tv_nsec = (unsigned long)(req->atime * 1000000) % 1000000 * 1000; \
+  }                                                                            \
+                                                                               \
+  if (!isnan(req->mtime)) {                                                    \
+    attr_list.commonattr |= ATTR_CMN_MODTIME;                                  \
+                                                                               \
+    times[i++].tv_sec = req->mtime;                                            \
+    times[i].tv_nsec = (unsigned long)(req->mtime * 1000000) % 1000000 * 1000; \
+  }
```

Comment:
Just so I can better understand this rule, there are already macros for `INIT`, `PATH`, `PATH2` and `POST` so how are they different than this new one?  I ask because I used those as the reasoning behind using a macro.  I'm completely cool with changing regardless but I wanted to better understand for next time.